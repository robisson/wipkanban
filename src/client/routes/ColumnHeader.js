// @flow
import * as React from "react";
import {withStyles} from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Typography from '@material-ui/core/Typography';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import brown from '@material-ui/core/colors/brown';
import MenuHeaderColumn from './MenuHeaderColumn';
import RemoveIcon from '@material-ui/icons/Remove';

const styles = () => ({
  root: {
    width: '100%',
    height: 35,
    backgroundColor: brown[300]
  },
  list: {
    paddingTop:5
  },
  listItem: {
    padding:0
  },
  listItemText: {
    margin:0,
    padding:0
  },
  listItemIconLeft: {
    marginRight:5,
    padding:0,
    color:'white',
    cursor:'pointer'
  }
});

type Props = {
  title:string,
  classes: Object
};

function ColumnHeader(props:Props) {

  const {title,classes} = props;

  return (
    <div className={classes.root}>
      <List className={classes.list}>
        <ListItem className={classes.listItem}>
          <ListItemIcon className={classes.listItemIconLeft}>
            <RemoveIcon/>
          </ListItemIcon>
          <ListItemText
            className={classes.listItemText}
            primary={< Typography component = "p" style = {{ color: '#FFFFFF' }} > {title} </Typography>}/>
          <ListItemSecondaryAction>
            <MenuHeaderColumn/>
          </ListItemSecondaryAction>
        </ListItem>
      </List>
    </div>
  );

}

export default withStyles(styles)(ColumnHeader);
