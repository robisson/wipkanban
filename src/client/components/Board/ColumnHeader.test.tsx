import React from "react";
import ColumnHeader from "./ColumnHeader";
import { createShallow } from "@material-ui/core/test-utils";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import { ShallowWrapper } from "enzyme";

describe("<ColumnHeader /> Component", () => {
  let shallow: any;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    wrapper = shallow(<ColumnHeader title="title" />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 3 items components", () => {
    expect(wrapper.find(ListItemText).type()).toEqual(ListItemText);
    expect(wrapper.find(ListItemIcon).type()).toEqual(ListItemIcon);
    expect(wrapper.find(ListItemSecondaryAction).type()).toEqual(
      ListItemSecondaryAction
    );
  });
});
