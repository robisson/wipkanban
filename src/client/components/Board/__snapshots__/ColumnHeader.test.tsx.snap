// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColumnHeader /> Component should render correctly 1`] = `undefined`;
